File: src/main/java/com/ata1635/cardgame/core/service/Deck.java
package com.ata1635.cardgame.core.service;

import com.ata1635.cardgame.core.model.Card;

import java.util.ArrayList;
import java.util.List;

public class Deck {

    private List<Card> deck;

    public Deck() {
        buildDeck();
    }

    public void buildDeck() {
        deck = new ArrayList<Card>();
        String[] values = {"A", "2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K"};
        String[] types = {"C", "D", "H", "S"};

        for (int i = 0; i < types.length; i++) {
            for (int j = 0; j < values.length; j++) {
                Card card = new Card(values[j], types[i]);
                deck.add(card);
            }
        }
    }

    public List<Card> getDeck() {
        return new ArrayList<Card>(deck);
    }

}

File: src/js/core/service/Deck.js
import { Card } from '../model/Card.js';

export class Deck {
    constructor() {
        this.deck = [];
        this.buildDeck();
    }

    getDeck() {
        return [...this.deck];
    }

    setDeck(cards) {
        this.deck = [...cards];
    }

    buildDeck() {
        const values = ["A", "2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K"];
        const types = ["C", "D", "H", "S"];
        for (let type of types) {
            for (let value of values) {
                this.deck.push(new Card(value, type));
            }
        }
    }

    shuffleDeck() {
        for (let i = this.deck.length - 1; i > 0; i--) {
            const j = Math.floor(Math.random() * (i + 1));
            [this.deck[i], this.deck[j]] = [this.deck[j], this.deck[i]];
        }
    }

    drawCard() {
        if (this.deck.length === 0) throw new Error("Deck is empty");
        return this.deck.pop();
    }
}